### PR TITLE
dfjr-814 implement featured menu content

### PR DESCRIPTION
**Pull request by anselmbradford**
## Additions

- Adds Featured Menu Content Molecule.

## Removals

- Old Featured Menu Content template.

## Changes

- Updates About Us Featured Menu Content image to be retina-display compatible.
 Also, removes play arrow.

## Testing

- `gulp build`
- Check menu and see that About Us and Policy & Compliance has Featured Menu Content.

## Review

- @jimmynotjim 
- @sebworks 
- @KimberlyMunoz 
- @sarahsimpson09 

## Screenshots

screen shot 2016-03-21 at 4 21 37 pm

screen shot 2016-03-21 at 4 21 29 pm

### CODE REVIEW

**Comment by sarahsimpson09** [comment]
Looks good to me. Thanks!

File: cfgov/jinja2/v1/_includes/molecules/featured-menu-content.html [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+   ==========================================================================
+
+   Description:
+
+   Create a featured content molecule. See
+   [GHE]/flapjack/Modules-V1/wiki/Featured-Menu-Content
+
+   value: An object with the following options for value.
+
+   value.body:  Body text.
```
Comment (jimmynotjim) [inline cfgov/jinja2/v1/_includes/molecules/featured-menu-content.html]: Minor, values should be left aligned

File: cfgov/jinja2/v1/_includes/molecules/featured-menu-content.html [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+   ========================================================================== #}
+
+{% macro render( value ) %}
+<a href="{{ value.link.url }}">
+    <img src="{{ value.image.src }}"
+         alt="{{ value.image.alt }}"
+         height="150"
+         width="270" />
+    <p class="h4 u-mt15">
```
Comment (jimmynotjim) [inline cfgov/jinja2/v1/_includes/molecules/featured-menu-content.html]: `u-mt15` can be removed, `h4` has 15px bottom by default now.

File: cfgov/jinja2/v1/_includes/molecules/featured-menu-content.html [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+   value.image.alt: A string for the image alt text.
+
+   ========================================================================== #}
+
+{% macro render( value ) %}
+<a href="{{ value.link.url }}">
+    <img src="{{ value.image.src }}"
+         alt="{{ value.image.alt }}"
+         height="150"
```
Comment (jimmynotjim) [inline cfgov/jinja2/v1/_includes/molecules/featured-menu-content.html]: Should we have height and width options to allow these to be changed?

File: cfgov/jinja2/v1/_includes/organisms/_vars-mega-menu.html [hunk 4]
```diff
@@ -19,11 +19,34 @@
...
+{% set poly_com_content = {
+    'link': {
+        'text': 'Resources to help you comply',
+        'url':  '/policy-compliance/guidance/implementation-guidance/',
+    },
+    'image': {
+        'src': '/static/img/fmc-poly-com-540x300.jpg',
+        'alt': 'Still from Know Before You Owe video.',
+    },
+    'body': 'The TILA-RESPA integrated disclosure rule replaces four disclosure forms with two new ones. We have resources to help you comply.'
```
Comment (jimmynotjim) [inline cfgov/jinja2/v1/_includes/organisms/_vars-mega-menu.html]: Add semantic breaks here and throughout.

File: cfgov/unprocessed/css/molecules/featured-menu-content.less [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    - name: Default example
+      markup: |
+          <aside class="m-featured-menu-content">
+              <a href="/the-bureau/">
+                  <img src="/static/img/fmc-about-us-540x300.jpg"
+                       alt="About Us video image"
+                       height="150"
+                       width="270">
+                  <p class="h4 u-mt15">
+                    The CFPB: Working for you
```
Comment (jimmynotjim) [inline cfgov/unprocessed/css/molecules/featured-menu-content.less]: Minor, update to four space indent

File: cfgov/unprocessed/css/molecules/featured-menu-content.less [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+  tags:
+    - cfgov-molecules
+*/
+
+.m-featured-menu-content {
+    .grid_column( 4 );
+
+    padding: 0 @grid_gutter-width / 2;
+
+    position: relative;
```
Comment (jimmynotjim) [inline cfgov/unprocessed/css/molecules/featured-menu-content.less]: Is this necessary?

File: cfgov/jinja2/v1/_includes/organisms/_vars-mega-menu.html (hunk 4 above)
Comment (sebworks) [inline cfgov/jinja2/v1/_includes/organisms/_vars-mega-menu.html]: I would love for this to be in Django with and provide access like `getConfig('MegaMenu')`.

**Comment by sebworks** [comment]
:+1:

**Comment by schaferjh** [comment]
@anselmbradford Design created another formation of the featured content module so it's not so tall -- see GHE/flapjack/Modules-V1/wiki/Featured-Menu-Content. I think that would be more suitable for the Policy & Compliance instance?

**Comment by anselmbradford** [comment]
@schaferjh ahh I see, oops overlooked that. I'll build it in. Would the image ever change dimensions, per @jimmynotjim comment?

File: cfgov/jinja2/v1/_includes/molecules/featured-menu-content.html [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+         width="270" />
```
Comment (KimberlyMunoz) [inline cfgov/jinja2/v1/_includes/molecules/featured-menu-content.html]: Closing the img tag is no longer required with an HTML5 doctype.

File: cfgov/unprocessed/css/molecules/featured-menu-content.less [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+              p.h4.u-mt15
+            p
+  tags:
+    - cfgov-molecules
+*/
+
+.m-featured-menu-content {
+    .grid_column( 4 );
+
+    padding: 0 @grid_gutter-width / 2;
```
Comment (KimberlyMunoz) [inline cfgov/unprocessed/css/molecules/featured-menu-content.less]: Should this be converted to ems?

File: cfgov/jinja2/v1/_includes/molecules/featured-menu-content.html (hunk 1 above)
Comment (anselmbradford) [inline cfgov/jinja2/v1/_includes/molecules/featured-menu-content.html]: I was aligning the same level e.g. (`value.body` and `value.link`; `value.link.url` and `value.link.text`), which we do sometimes elsewhere, but there's space for them all to align here, will fix. Also, actually I didn't align body and link :stuck_out_tongue:

File: cfgov/jinja2/v1/_includes/molecules/featured-menu-content.html (hunk 3 above)
Comment (anselmbradford) [inline cfgov/jinja2/v1/_includes/molecules/featured-menu-content.html]: See comment below to Jess. Checking if they will ever change dimensions.

**Comment by schaferjh** [comment]
@anselmbradford I don't think so...it should just disappear once the space gets too small.

File: cfgov/jinja2/v1/_includes/molecules/featured-menu-content.html (hunk 1 above)
Comment (anselmbradford) [inline cfgov/jinja2/v1/_includes/molecules/featured-menu-content.html]: Fixed in https://github.com/cfpb/cfgov-refresh/pull/1665/commits/60b8eaf28eafee20484c08edd6d6aeb6fb37dce1

File: cfgov/jinja2/v1/_includes/molecules/featured-menu-content.html (hunk 2 above)
Comment (anselmbradford) [inline cfgov/jinja2/v1/_includes/molecules/featured-menu-content.html]: Fixed in https://github.com/cfpb/cfgov-refresh/pull/1665/commits/60b8eaf28eafee20484c08edd6d6aeb6fb37dce1

File: cfgov/jinja2/v1/_includes/organisms/_vars-mega-menu.html (hunk 4 above)
Comment (anselmbradford) [inline cfgov/jinja2/v1/_includes/organisms/_vars-mega-menu.html]: Fixed in https://github.com/cfpb/cfgov-refresh/pull/1665/commits/60b8eaf28eafee20484c08edd6d6aeb6fb37dce1

File: cfgov/unprocessed/css/molecules/featured-menu-content.less (hunk 6 above)
Comment (anselmbradford) [inline cfgov/unprocessed/css/molecules/featured-menu-content.less]: Removed in https://github.com/cfpb/cfgov-refresh/pull/1665/commits/60b8eaf28eafee20484c08edd6d6aeb6fb37dce1

File: cfgov/jinja2/v1/_includes/molecules/featured-menu-content.html (hunk 7 above)
Comment (anselmbradford) [inline cfgov/jinja2/v1/_includes/molecules/featured-menu-content.html]: Removed in https://github.com/cfpb/cfgov-refresh/pull/1665/commits/60b8eaf28eafee20484c08edd6d6aeb6fb37dce1

File: cfgov/unprocessed/css/molecules/featured-menu-content.less (hunk 8 above)
Comment (anselmbradford) [inline cfgov/unprocessed/css/molecules/featured-menu-content.less]: Converted in https://github.com/cfpb/cfgov-refresh/pull/1665/commits/60b8eaf28eafee20484c08edd6d6aeb6fb37dce1

**Comment by schaferjh** [comment]
@sarahsimpson09 In the content guidelines for the featured menu content, we said the description should be limited to one sentence. Since the second sentence in the content here is a repeat of the call to action, do you think we could cut it? I'm worried by how tall the content is when the menu options are relatively few. (It appears that a horizontal configuration will need to be developed post launch)

Also, any concerns around the fact that the link does not actually go to a page within Policy & Compliance? And the page it links to doesn't reference the Policy & Compliance section in any way?

File: cfgov/unprocessed/css/molecules/featured-menu-content.less (hunk 5 above)
Comment (anselmbradford) [inline cfgov/unprocessed/css/molecules/featured-menu-content.less]: Fixed

File: cfgov/jinja2/v1/_includes/molecules/featured-menu-content.html (hunk 3 above)
Comment (anselmbradford) [inline cfgov/jinja2/v1/_includes/molecules/featured-menu-content.html]: Sounds like the image would retain the same size.

**Comment by anselmbradford** [comment]
Ready to review again (aside from content updates). Added:
- horizontal layout

File: cfgov/unprocessed/css/molecules/featured-menu-content.less (hunk 8 above)
Comment (anselmbradford) [inline cfgov/unprocessed/css/molecules/featured-menu-content.less]: @jimmynotjim can you confirm that I converted these correctly?

File: cfgov/jinja2/v1/_includes/molecules/featured-menu-content.html (hunk 3 above)
Comment (jimmynotjim) [inline cfgov/jinja2/v1/_includes/molecules/featured-menu-content.html]: :ok_hand:

File: cfgov/unprocessed/css/molecules/featured-menu-content.less (hunk 8 above)
Comment (jimmynotjim) [inline cfgov/unprocessed/css/molecules/featured-menu-content.less]: Yep, the font-size hasn't been changed here so you're fine to use the `@base-font-size-px` var.

**Comment by jimmynotjim** [comment]
:+1:

**Comment by sebworks** [comment]
:+1:

**Pull request merged**